src/sentry/static/sentry/app/components/forms/selectControl.tsx

```diff
@@ -48,13 +57,36 @@ const combineStyles = (newStyles, baseStyles) => {
   }, baseStyles);
 };
 
-const SelectControl = props => {
+type ControlProps = React.ComponentProps<typeof ReactSelect> & {
+  theme: Theme;
+  /**
+   * Ref forwarded into ReactSelect component.
+   * The any is inherited from react-select.
+   */
+  forwardedRef: React.Ref<ReactSelect>;
+  /**
+   * Set to true to prefix selected values with content
+   */
+  inFieldLabel?: string;
+  /**
+   * Backwards compatible shim to work with select2 style choice type.
+   */
+  choices?: Choices | ((props: ControlProps) => Choices);
+  /**
+   * Use react-select v2. Deprecated, don't make more of this.
+   */
+  deprecatedSelectControl?: boolean;
+};
+
+type LegacyProps = React.ComponentProps<typeof SelectControlLegacy>;
+
+function SelectControl(props: ControlProps) {
```

Comment:
I believe ReactSelect lets you use generics for the values. Can we bring that in too?

Reply:
I can give it a shot. Hopefully the styled components, HOC and forwarded ref don't interfere.

Reply:
I'm going to revisit adding generics when I update the types package. We currently get basic types for the select options (label, value). While we  have a few select controls that extend that interface, they don't remove from label/value.